### ISSUE TITLE
UW2 not saving correctly

### ISSUE THREAD

**Issue by KarlClinckspoor** [hankmorgan/Underworld-Editor#3]
Hey hank, I hope you're doing well. When you get back to these projects, hopefully you'll take a look at this.

I was messing around with this editor. It's quite cool, and it can load UW2 too, as opposed to the one made by krokots for example. However, when I edit a UW2 lev.ark and save it, the map data gets corrupted. Here's what I did to trigger this bug.

1. Load a fresh 1993 save0 lev.ark.
2. Open block #0 and block # 1 tilemaps to check they're working.
3. Edit the ankh rug in the Avatar's quarters to a solid tile.
4. Save lev.ark
5. Reload lev.ark
6. Notice the ankh rug tile is back. Access block # 1 tilemap, you'll enter an infinite while loop, consuming more and more memory until the program crashes.

I *think* this has something to do with the compressed nature of UW2 lev.ark and modifying a byte in the wrong place. Here's what I did to load it correctly:

1. Load a fresh 1993 save0 lev.ark.
2. Repack lev.ark
3. Modify ankh for something else.
4. Save
5. Reload lev.ark
6. Now the modification was kept and I can load block # 1, but if I load this file in a fresh UW2 journey, it loads incorrectly.

**Comment by hankmorgan** [comment]
Yes. Saving of a compressed UW2 file does not work since I do not have a compression algorithm yet. Editing of Underworld 2 only works with a special decompressed lev.ark file that stores the first 80 map blocks uncompressed. There is an option in the menu to create this file that can then be edited. Look for the option Repack UW2 Lev.ark.  The Dos game accepts this file but does not like it if more than 80 blocks are decompressed due to memory constraints.

Obviously back up the 1993 file first.

**Comment by hankmorgan** [comment]
reading a bit closer at what you've said re the crash. I think those bugs have been fixed in more recent versions of the editor. I'm actively refreshing the editor currently to assist in game research so I think those issues are resolved. However I would recommend creating the unpacked lev.ark, creating a new game with that file in the data folder and then saving to a safe slot for editing. 

A quick check I found that tells me if the file is faulty or not is to change levels in game. If you can change level okay without a crash the file is probably okay.
